### PR TITLE
Feat 951 rig session compatibility

### LINKED ISSUE #951: Rig and session compatibility check
# User story
As a user, I want to check that a rig.json and session.json are compatible. 

# Acceptance criteria
- [ ] Given jsons, extract models
- [ ] Given rig and session models, compare rig_id and mouse_platform_name
- [ ] Given rig and session models, compare stream data
- [ ] Given rig and session models, extract and compare stimulus epochs

# Sprint Ready Checklist 
- [ ] 1. Acceptance criteria defined 
- [ ] 2. Team understands acceptance criteria 
- [ ] 3. Team has defined solution / steps to satisfy acceptance criteria 
- [ ] 4. Acceptance criteria is verifiable / testable 
- [ ] 5. External / 3rd Party dependencies identified
- [ ] 6. Ticket is prioritized and sized

# Notes

### PR DESCRIPTION
**Pull request by mekhlakapoor**
closes #951 

Checks whether mouse platform name and rig id match in session and rig models. If rig_id doesn't match, raises ValueError. Once we add this as a validator in Metadata, we can change it to raise ValidationError

### CODE REVIEW

File: src/aind_data_schema/utils/compatibility_check.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        """Compares mouse_platform_name"""
+        return self.session.mouse_platform_name == self.rig.mouse_platform.name
+
+    def run_compatibility_check(self) -> Optional[Dict]:
+        """Runs compatibility check.Creates a dictionary of fields and whether it matches in rig and session."""
+        if self.compare_rig_id() is False:
+            raise ValueError(f"Rig ID in session {self.session.rig_id} does not match the rig {self.rig.rig_id}.")
+        else:
+            comparison_data = {
+                "rig_id": self.compare_rig_id(),
```
Comment (jtyoung84) [inline src/aind_data_schema/utils/compatibility_check.py:30]: Maintaining these key names might be troublesome. Maybe just have the `compare_` methods return `Optional[ValueError]` and have `run_compatibility_check` create a `List[ValueError]` and raise an error if that list is non-empty. It should return `None`.

Comment (mekhlakapoor) [reply src/aind_data_schema/utils/compatibility_check.py:30]: would we want to return something to users that indicates which fields are not synced though?

Comment (jtyoung84) [reply src/aind_data_schema/utils/compatibility_check.py]: Aren't those in the error messages?

Comment (mekhlakapoor) [reply src/aind_data_schema/utils/compatibility_check.py]: ya so nvm on my previous comment

**Review by saskiad — APPROVED** [review]

**Pull request merged**